bin/hdfcoinc/pycbc_multiifo_combine_statmap

```diff
@@ -0,0 +1,182 @@
+for ifo in args.ifos:
+    all_trig_times[ifo] = numpy.array([], dtype=numpy.uint32)
+    all_trig_ids[ifo] = numpy.array([], dtype=numpy.uint32)
+
+# for each file, append the trigger time and id data for each ifo
+#   - if it isnt there, then fill with -1 values
+for f_in in files:
+    for ifo in args.ifos:
+        if ifo in f_in['foreground']:
+            all_trig_times[ifo] = numpy.concatenate([all_trig_times[ifo], \
+                                     f_in['foreground/{}/time'.format(ifo)][:]])
+            all_trig_ids[ifo] = numpy.concatenate([all_trig_ids[ifo], 
+                                     f_in['foreground/{}/trigger_id'.format(ifo)][:]])
+        else:
+            all_trig_times[ifo] = numpy.concatenate([all_trig_times[ifo],
+                                     -1*numpy.ones_like(f_in['foreground/fap'][:], dtype=numpy.uint32)])
+            all_trig_ids[ifo] = numpy.concatenate([all_trig_ids[ifo],
+                                     -1*numpy.ones_like(f_in['foreground/fap'][:], dtype=numpy.uint32)])
+
+logging.info('clustering triggers for loudest stat value')
+#cluster the triggers before applying trials factor
+for ifo in args.ifos:
+    f['foreground/{}/time'.format(ifo)] = all_trig_times[ifo]
+    f['foreground/{}/trigger_id'.format(ifo)] = all_trig_ids[ifo]
+
+#cluster for loudest stat value 
+def argmax(v):
+    return numpy.argsort(v)[-1]
+
+stat = numpy.core.records.fromarrays([f['foreground/ifar'][:],
+                                      f['foreground/stat'][:]],
+                                      names='ifar,stat')
+
+all_times = ()
+for ifo in args.ifos:
+    all_times = all_times +  (f['foreground/%s/time' % ifo][:],)
+
+cidx = pycbc.events.cluster_coincs_multiifo(stat, all_times,
+                                            numpy.zeros(len(stat)), 0,
+                                            args.cluster_window, argmax)
+
+# downsample the foreground columns to only the loudest ifar between the multiple files
+for key in f['foreground'].keys():
+    if key not in args.ifos:
+        dset = f['foreground/%s' % key][:][cidx]
+        del f['foreground/%s' % key]
+        f['foreground/%s' % key] = dset
+    else: # key is an ifo
+        for k in  f['foreground/%s' % key].keys():
+            dset = f['foreground/{}/{}'.format(key,k)][:][cidx]
+            del f['foreground/{}/{}'.format(key,k)]
+            f['foreground/{}/{}'.format(key,k)] = dset
+
+logging.info('applying trials factor')
+#copy across ifar and ifar_exc, etc 
+# recalculate ifar/fap for the foreground triggers by 
+#  applying a trials factor = num_files(num_bins)
+
+all_times = ()
+for ifo in args.ifos:
+    all_times = all_times +  (f['foreground/%s/time' % ifo][:],)
+
+# trials factor is how many possible 2+IFO combinations are 'on' at the time of the coincidence
+trials_factors = numpy.zeros_like(f['foreground/ifar'][:])
+test_times = numpy.array([pycbc.events.mean_if_greater_than_zero(tc)[0] for tc in zip(*all_times)])
```

Comment:
i'm not following what `zip(*...)` does here.  isn't 'all_times' just a single sequence?

Reply:
all_times is a tuple of lists of the times in each detector, so need to unpack the tuple using * in order to zip them back together